test/library/packages/UnitTest/TestResult.chpl

```diff
@@ -0,0 +1,123 @@
+/*
+  Holder for test result information.
+  Test results are automatically managed by the TestRunner, and do not 
+  need to be explicitly manipulated by writers of tests.  
+  Each instance holds the total number of tests run, and collections of 
+  failures and errors that occurred among those test runs. The collections
+  contain tuples of (testcase, exceptioninfo), where exceptioninfo is the
+  formatted traceback of the error that occurred.
+*/
+class TestResult {
+  type tupType = 2*string;
+  var failures: [1..0] tupType,
+      errors: [1..0] tupType,
+      skipped: [1..0] tupType;
+  var testsRun = 0;
+  var shouldStop = false;
+  var separator1 = "="* 70,
+      separator2 = "-"* 70;
+  // called when a test if about to run
+  proc startTest() {
+    this.testsRun += 1;
+  }
+  
+  /*Called when an error has occurred.*/
+  proc addError(test, errMsg) {
```

Comment:
I'd suggest concrete types for the public interface functions here (assuming they don't need to be generic), e.g.

```chpl
proc addError(test: string, errMsg: string) { .. }
```

I would expect that we can drop the casts in the push_back call then.

Reply:
I have made these modification.